CoastalResilienceNetwork/GeositeFramework
# Demonstrate using modal image in sample plugin

## Overview

This PR updates the sample `identify_point` plugin to include a demonstration of how to create an image modal popup triggered on clicking a thumbnail. I tried to keep the footprint of the changes as small as possible; the updates here include:

- create a clickable div for the thumbnail in the plugin
- add a click event listener to the div which calls a new `renderSampleImageModal` method
- use TINY.box to render a modal from the image source path passed in as an argument to `renderSampleImageModal`

Connects #888 

## Screenshots

*Thumbnail*
<img width="749" alt="screen shot 2017-02-21 at 1 14 51 pm" src="https://cloud.githubusercontent.com/assets/4165523/23178453/728c69ec-f838-11e6-91d0-5bbf69444f16.png">

*Modal*
<img width="882" alt="screen shot 2017-02-21 at 1 14 58 pm" src="https://cloud.githubusercontent.com/assets/4165523/23178460/7c272ae6-f838-11e6-8e38-ff1218fd7621.png">

## Notes

I tried to keep the changes here small but legible, and I'm happy to push it more in one direction or the other. Once this is ready to merge, I'll update the Wiki either to add a new page or to update the "Plugin Settings" page: https://github.com/CoastalResilienceNetwork/GeositeFramework/wiki/Plugin-Settings

## Testing

 * rebuild this branch in VS, then view it in the browser
 * open the "Identify Point" plugin and click on the thumbnail
 * verify that you see the modal appear on the screen
 * click the background to close the modal and verify that it disappears
